app/models/api/sample_io.rb

```diff
@@ -68,12 +68,47 @@ def json_root
     map_attribute_to_json_attribute(:supplier_name)
     map_attribute_to_json_attribute(:donor_id)
     map_attribute_to_json_attribute(:phenotype)
+    map_attribute_to_json_attribute(:sibling)
+    map_attribute_to_json_attribute(:is_resubmitted)
+    map_attribute_to_json_attribute(:date_of_sample_collection)
+    map_attribute_to_json_attribute(:date_of_sample_extraction)
+    map_attribute_to_json_attribute(:sample_extraction_method)
+    map_attribute_to_json_attribute(:sample_purified)
+    map_attribute_to_json_attribute(:purification_method)
+    map_attribute_to_json_attribute(:concentration)
+    map_attribute_to_json_attribute(:concentration_determined_by)
+    map_attribute_to_json_attribute(:sample_type)
+    map_attribute_to_json_attribute(:sample_storage_conditions)
+    map_attribute_to_json_attribute(:genotype)
+    map_attribute_to_json_attribute(:age)
+    map_attribute_to_json_attribute(:cell_type)
+    map_attribute_to_json_attribute(:disease_state)
+    map_attribute_to_json_attribute(:compound)
+    map_attribute_to_json_attribute(:dose)
+    map_attribute_to_json_attribute(:immunoprecipitate)
+    map_attribute_to_json_attribute(:growth_condition)
+    map_attribute_to_json_attribute(:rnai)
+    map_attribute_to_json_attribute(:organism_part)
+    map_attribute_to_json_attribute(:time_point)
+    map_attribute_to_json_attribute(:disease)
+    map_attribute_to_json_attribute(:subject)
+    map_attribute_to_json_attribute(:treatment)
+    map_attribute_to_json_attribute(:genome_size)
+    map_attribute_to_json_attribute(:saphyr)
+    map_attribute_to_json_attribute(:pacbio)
+    map_attribute_to_json_attribute(:date_of_consent_withdrawn)
+    map_attribute_to_json_attribute(:user_id_of_consent_withdrawn)
   end
 
   extra_json_attributes do |_object, json_attributes|
     if json_attributes['reference_genome'].blank?
       json_attributes['reference_genome'] = nil
     end
+
+    user_id = json_attributes['user_id_of_consent_withdrawn']
+    if user_id.present? && User.find(user_id)
+      json_attributes['user_id_of_consent_withdrawn'] = User.find(user_id)&.login
+    end
```

Comment:
User.find(x) raises ActiveRecord::RecordNotFound if the user is missing.

Instead this should probably be:
```ruby
   if user_id.present?
      json_attributes['user_id_of_consent_withdrawn'] = User.find_by(id: user_id)&.login
    end
```
find_by(id:x) returns nil if not found. This approach also gets rid of the double lookup of the user.

Reply:
Good spot thanks